docs/architecture/deep-dives/autofill/autofill-menu.md

```diff
@@ -0,0 +1,299 @@
+
+The user interface representing the Auto-fill Menu consists of extension pages rendered within
+sandboxed iframes. These sandboxed iframes are structured within the custom elements injected into a
+user's webpage and represent the Auto-fill Menu
+[button](https://github.com/bitwarden/clients/tree/main/apps/browser/src/autofill/overlay/pages/button)
+and
+[list](https://github.com/bitwarden/clients/tree/main/apps/browser/src/autofill/overlay/pages/list)
+UI elements.
+
+Rendering these views within a sandboxed iframe sets clear limitations on how these pages can
+communicate with the extension background. This approach effectively
+[strips the pages of the ability to directly access the extension API](https://developer.chrome.com/docs/extensions/mv3/sandboxingEval/),
+necessitating communication with the extension background through postMessage calls routed through
+the iframe element's parent. This ensures that any actions needing to occur within the extension
+background script cannot be initiated outside the
+[isolated context of the content script](https://developer.chrome.com/docs/extensions/mv3/content_scripts/#isolated_world)
+that injects the Auto-fill Menu UI elements into the DOM.
+
+These pages are rendered with a strict content security policy that prohibits the execution of any
+inline scripts or scripts foreign to the extension. This ensures that the pages cannot execute any
+form of script not explicitly defined within the extension, whether inline or through a script tag.
+
+### Passing Messages Between the Auto-fill Menu and the Extension
+
+As a consequence of the sandboxed iframe implementation, an indirect messaging system was necessary
+to enable communication between the extension background and the Auto-fill Menu pages. This
+messaging approach operates through the use of `postMessage` calls exchanged among the Auto-fill
+Menu pages, the content script, and the extension background.
+
+When an Auto-fill Menu page needs to communicate with the extension background, it initiates a
+`postMessage` call to its parent frame, the content script that injected the Auto-fill Menu into the
+webpage. This content script verifies the origin of the window message to confirm it originates from
+the expected source. Upon validation, it relays a message to the extension background via a
+long-lived port connection. The extension background, upon receiving this message, executes the
+necessary background logic and responds with a port message back to the content script. This message
+is then transferred to the Auto-fill Menu page through another `postMessage` call.
+
+Below is an illustration that demonstrates how this messaging system operates when initializing and
+retrieving cipher data for the Auto-fill Menu list page:
+
+[![Auto-fill Menu messaging](./assets/autofill-overlay-messaging.jpg) _(A visual representation of the messaging workflow of the Auto-fill Menu)_](./assets/autofill-overlay-messaging.jpg)
```

Comment:
❓ Are these diagrams' editable versions stored somewhere for us internally? Is it possible to represent these as SVG files or even (Plant)UML so we don't need to maintain something elsewhere and the content is all here?

Reply:
Ah yes, they are editable, I created these within Excalidraw. 

I originally set these as JPGs because the cursive font created fairly large SVG files... however with the change to sans-serif fonts, the SVGs are much smaller than the JPGs. I've updated these files to present as SVG elements.

Also, I've added in the editable Excalidraw file to facilitate updates in the future as necessary.